packages/services/src/infrastructure/api/leather/leather-api.client.ts

```diff
@@ -60,65 +51,69 @@ export class LeatherApiClient {
     descriptor: string,
     pageRequest: LeatherApiPageRequest,
     signal?: AbortSignal
-  ): Promise<LeatherApiPage<LeatherApiBitcoinTransaction>> {
+  ) {
     const params = getPageRequestQueryParams(pageRequest);
-    params.append('network', selectBitcoinNetwork(this.settingsService.getSettings()));
+    const network = selectBitcoinNetwork(this.settingsService.getSettings());
     return await this.cacheService.fetchWithCache(
       ['leather-api-transactions', descriptor, params.toString()],
       async () => {
-        const res = await axios.get<LeatherApiPage<LeatherApiBitcoinTransaction>>(
-          `${LEATHER_API_URL}/v1/transactions/${descriptor}?${params.toString()}`,
-          { signal }
-        );
-        return leatherApiBitcoinTransactionPageSchema.parse(res.data);
+        const { data, error } = await this.client.GET(`/v1/transactions/{descriptor}`, {
+          params: {
+            path: { descriptor },
+            query: {
+              network,
+              page: pageRequest.page.toString(),
+              pageSize: pageRequest.pageSize.toString(),
+            },
+          },
+          signal,
+        });
+        if (error) {
+          throw new Error(error.error);
+        }
+        return data;
       },
       { ttl: HttpCacheTimeMs.fiveSeconds }
     );
   }
 
-  async fetchFiatExchangeRates(signal?: AbortSignal): Promise<LeatherApiFiatRates> {
+  async fetchFiatExchangeRates(signal?: AbortSignal) {
     return await this.cacheService.fetchWithCache(
       ['leather-api-fiat-rates'],
       async () => {
-        const res = await axios.get<LeatherApiFiatRates>(
-          `${LEATHER_API_URL}/v1/market/fiat-rates`,
-          {
-            signal,
-          }
-        );
-        return leatherApiFiatRatesSchema.parse(res.data);
+        const { data, error } = await this.client.GET('/v1/market/fiat-rates', { signal });
+        if (error) {
+          throw new Error(error.error);
+        }
+        return data;
       },
       { ttl: HttpCacheTimeMs.oneDay }
     );
   }
 
-  async fetchCryptoPrices(signal?: AbortSignal): Promise<LeatherApiCryptoPrices> {
+  async fetchCryptoPrices(signal?: AbortSignal) {
     return await this.cacheService.fetchWithCache(
       ['leather-api-crypto-prices'],
       async () => {
-        const res = await axios.get<LeatherApiCryptoPrices>(
-          `${LEATHER_API_URL}/v1/market/crypto-prices`,
-          {
-            signal,
-          }
-        );
-        return leatherApiCryptoPricesSchema.parse(res.data);
+        const { data, error } = await this.client.GET('/v1/market/crypto-prices', { signal });
+        if (error) {
+          throw new Error(error.error);
+        }
```

Comment:
Worth making a wrapper to not have to repeat throwing the error? And, doesn't the generated client have an option to throw rather than safely return?